lxd/cgroup/init.go

```diff
@@ -72,3 +87,11 @@ func init() {
 		}
 	}
 }
+
+func WantsMountCgroupsForce() bool {
+	return cgFullCgroup2 && cgCgroupNamespace
+}
+
+func WantsMountCgroupsMixed() bool {
+	return cgCgroupNamespace
+}
```

Comment:
This is too lxc specific.

What we could introduce is something like:
```go
// Info contains system cgroup information
type Info struct {
    // Mode is one of "disabled", "cgroup1", "cgroup2" or "hybrid"
    Mode string

    // Namespacing indicates support for the cgroup namespace
    Namespacing bool
}

func GetInfo() Info {
    info := Info{}
    info.Namespacing = cgNamespace
    info.Mode = ...

    return info
}
```

This can then later be extended to also give us details on all the controllers and be used to show the status on LXD startup.

Reply:
> Reading through the cgroup detection logic you sent yesterday. That seems broken. The code doesn't seem to be able to detect a pure cgroup2 layout.

Hm, I see it's a fallback option...